### PR TITLE
makefile: update operator-sdk version

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

**Comment by malayparida2000** [comment]
@iamniting We also need to update a lot of things following this
https://sdk.operatorframework.io/docs/upgrading-sdk-version/,
From 1.14.0 we have to look at each version and the recommended changes. Especially we have to make the changes 
recommended for config/ folder & other changes also.

**Comment by iamniting** [comment]
> @iamniting We also need to update a lot of things following this https://sdk.operatorframework.io/docs/upgrading-sdk-version/, From 1.14.0 we have to look at each version and the recommended changes. Especially we have to make the changes recommended for config/ folder & other changes also.

done

File: hack/make-tools.mk [hunk 1]
```diff
@@ -7,7 +7,7 @@ endef
 
 CONTROLLER_GEN = $(BIN_DIR)/controller-gen
 controller-gen: ## Download controller-gen locally if necessary.
-	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
+	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.0)
```
Comment (malayparida2000) [inline hack/make-tools.mk:10]: https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.23.0/#gov3-update-controller-tools-from-090-to-092, This needs to be 0.9.2

**Comment by malayparida2000** [comment]
We are using a different image here, can you check if it's okay
https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.23.0/#upgrade-kube-proxy-image-gcriokubebuilderkube-rbac-proxy-from-v0110-to-v0130

**Comment by malayparida2000** [comment]
We are still missing
https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.17.0/#add-resource-requests-and-limits-to-kube-rbac-proxy
https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.23.0/#ansiblev1helmv1-upgrade-kustomize-versions-from-v387-to-v455

**Comment by malayparida2000** [comment]
Is this meaningful to us https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.28.0/#upgrade-k8s-versions-to-use-126

File: hack/make-tools.mk (hunk 1 above)
Comment (iamniting) [reply hack/make-tools.mk]: done

**Comment by iamniting** [comment]
> We are using a different image here, can you check if it's okay https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.23.0/#upgrade-kube-proxy-image-gcriokubebuilderkube-rbac-proxy-from-v0110-to-v0130

updated rbac-roxy image.

**Comment by iamniting** [comment]
> We are still missing https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.17.0/#add-resource-requests-and-limits-to-kube-rbac-proxy 

We don't need this

> https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.23.0/#ansiblev1helmv1-upgrade-kustomize-versions-from-v387-to-v455

Updated

**Review by malayparida2000 — APPROVED** [review]
/lgtm

**Pull request merged**